src/Google/GlobalSiteTag.php

```diff
@@ -357,62 +357,64 @@ public function maybe_display_conversion_and_purchase_event_snippets( string $ad
 		);
 		wp_print_inline_script_tag( $purchase_page_gtag );
 
-		// Enhanced conversion data.
-		$ec_data         = [];
-		$customer        = new WC_Customer( $order->get_customer_id() );
-		$email           = $customer->get_billing_email();
-		$fname           = $customer->get_billing_first_name();
-		$lname           = $customer->get_billing_last_name();
-		$phone           = $customer->get_billing_phone();
-		$billing_address = $customer->get_billing_address();
-		$postcode        = $customer->get_billing_postcode();
-		$city            = $customer->get_billing_city();
-		$region          = $customer->get_billing_state();
-		$country         = $customer->get_billing_country();
-
-		// Add email in EC data.
-		if ( ! empty( $email ) ) {
-			$ec_data['email'] = $email;
-		}
+		if ( $this->is_enhanced_conversion_enabled() ) {
+			// Enhanced conversion data.
+			$ec_data         = [];
+			$customer        = new WC_Customer( $order->get_customer_id() );
+			$email           = $customer->get_billing_email();
+			$fname           = $customer->get_billing_first_name();
+			$lname           = $customer->get_billing_last_name();
+			$phone           = $customer->get_billing_phone();
+			$billing_address = $customer->get_billing_address();
+			$postcode        = $customer->get_billing_postcode();
+			$city            = $customer->get_billing_city();
+			$region          = $customer->get_billing_state();
+			$country         = $customer->get_billing_country();
+
+			// Add email in EC data.
+			if ( ! empty( $email ) ) {
+				$ec_data['email'] = $this->normalize_and_hash( $email );
```

Comment:
Make sure all hashed data is saved using the expected keys described in [these instructions](https://support.google.com/google-ads/answer/13262500?hl=en#zippy=identify-and-define-your-enhanced-conversions-variables%2Cset-up-standard-automatic-enhanced-conversions).

For example, this should be:

```suggestion
				$ec_data['sha256_email_address'] = $this->normalize_and_hash( $email );
```

🔢 Applies here and to the other gtag values.